test/babel-config.js

```diff
@@ -9,6 +9,14 @@ const configManager = require('hullabaloo-config-manager');
 const babelConfigHelper = require('../lib/babel-config');
 
 const fixture = name => path.join(__dirname, 'fixture', name);
+const setNodeVersion = value => Object.defineProperty(process.versions, 'node', {value});
+const resetNodeVersion = setNodeVersion.bind(null, process.versions.node);
+
+// Execute fn with a given stubbed node version, then reset to the real version
+const withNodeVersion = (version, fn) => {
+	setNodeVersion(version);
+	return Promise.resolve(fn).then(() => resetNodeVersion());
```

Comment:
You never actually execute `fn` though 😉 

I'd return the result of `fn` which makes avoids some odd indentation in the tests themselves. For example:

```js
function withNodeVersion (version, run) {
  setNodeVersion(version);
  const promise = new Promise(resolve => {
    resolve(run());
  });
  promise.then(resetNodeVersion, resetNodeVersion);
  return promise;
}
```